### ISSUE TITLE
Dependency issue with development installation

### ISSUE THREAD

**Issue by Aratz** [mamba-org/mamba#841]
Hi!

I've been trying to build the last version of Mamba on a fresh Ubuntu 20.04 VM, following the instructions in the readme. When running `pip install -e .`, I run into the following error:
```
    src/core/repo.cpp: In member function 'bool mamba::MRepo::read_file(const string&)':
    src/core/repo.cpp:247:60: error: 'CONDA_ADD_USE_ONLY_TAR_BZ2' was not declared in this scope
      247 |         int flags = Context::instance().use_only_tar_bz2 ? CONDA_ADD_USE_ONLY_TAR_BZ2 : 0;
          |                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
    error: command 'gcc' failed with exit status 1
```

Somehow it seems related to 748060c. When running `mamba install ... libsolv ... -c conda-forge` from the readme, version 0.7.16 is installed on my machine but version >=0.7.18 is required since that commit.

**Comment by adriendelsalle** [comment]
Hi @Aratz 

Did you already had `libsolv` installed in the env where you ran `mamba install ... libsolv ... -c conda-forge` ?
If yes, it's normal behavior that it's not updated, since you required installation and it's already installed.
You have to run `mamba update libsolv -c conda-forge` to update it.

But thanks for the pointer of updating the `environment-dev.yml` yaml file and the PR :)

**Comment by Aratz** [comment]
I just redid the installation process on a fresh VM to understand what happened:

1.  I installed Anaconda with the installer from the official website, at that point `libsolv` was not installed
2. I installed mamba (with `conda install mamba -n base -c conda-forge`, as specified in the readme). `libsolv-0.7.16` is installed then, although I don't quite understand why it's not 0.7.18 :thinking: 
3. Then I ran `mamba install ... libsolv ... -c conda-forge`, but as you said, `libsolv` is not updated then.

So it seems it's something to do with how conda solves the environment. With the regular conda, it really takes a lot of time and it seems that conda fails several times before it manages to solve the dependencies.

PS: I just tried with Miniconda, which has a much smaller base, solving the environment is much faster and `libsolv` 0.7.18 is installed immediately then.
